### PR TITLE
Use existing cryptsetup API for changing keyslot passphrase

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Instead of manually removing the keyslot and adding new a one.
Our old code also doesn't work in FIPS mode.

### CODE REVIEW

File: src/plugins/crypto.c [hunk 1]
```diff
@@ -1394,41 +1392,17 @@ gboolean bd_crypto_luks_change_key_blob (const gchar *device, const guint8 *pass
...
-        crypt_free (cd);
-        g_free (volume_key);
-        bd_utils_report_finished (progress_id, (*error)->message);
-        return FALSE;
-    }
-
-    ret = crypt_keyslot_add_by_volume_key (cd, ret, volume_key, vk_size, (char*) npass_data, ndata_len);
+    ret = crypt_keyslot_change_by_passphrase (cd, CRYPT_ANY_SLOT, CRYPT_ANY_SLOT,
+                                              (char*) pass_data, data_len,
+                                              (char*) npass_data, ndata_len);
```
Comment (vpodzime) [inline src/plugins/crypto.c]: One disadvantage of this is that we lose the information about which phase failed. What is the error we get when the old passphrase is incorrect?

Comment (vojtechtrefny) [reply src/plugins/crypto.c]: It returns `EPERM` so just the standard "Operation not permitted". But more information should be available in the log (and we redirect cryptsetup log to our log, so it should be enough).

Comment (vpodzime) [reply src/plugins/crypto.c]: Aren't we processing `EPERM` in a special way somewhere else in this plugin?

Comment (vojtechtrefny) [reply src/plugins/crypto.c]: Good point. I've added custom error message.

**Pull request merged**